### PR TITLE
🐛 [RUMF-1372] Should not set `rule_psr` if it is not defined by the user

### PR DESCRIPTION
**Pull request by liywjl**
## Motivation

We make a distinction between user set settings and default settings. By sending `rule_psr` even when the user has not set it, we loose that distinction. 

## Changes

Set `rule_psr` to undefined when it is not present, thus omiting it during stringification

## Testing

- [ ] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts [hunk 1]
```diff
@@ -167,5 +167,5 @@ function toPerformanceEntryRepresentation(entry: RumPerformanceEntry): Performan
  * @returns number between 0 and 1 which represents tracing sample rate
  */
 function getRulePsr(configuration: RumConfiguration) {
-  return configuration.tracingSampleRate / 100
+  return configuration.tracingSampleRate ? configuration.tracingSampleRate / 100 : undefined
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:170]: **❗ warning:** ‏a tracing sample rate set to `0` would return `undefined`

Comment (liywjl) [reply packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:170]: **❓ question:** is tracing sample rate a valid input?

Comment (bcaudan) [reply packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:170]: https://docs.datadoghq.com/real_user_monitoring/browser/#initialization-parameters
Capture d’écran 2022-08-26 à 10 06 35

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.spec.ts [hunk 2]
```diff
@@ -215,6 +215,55 @@ describe('resourceCollection', () => {
...
+          traceSampled: true,
+          spanId: new TraceIdentifier(),
+          traceId: new TraceIdentifier(),
+        })
+      )
+      const traceInfo = (rawRumEvents[0].rawRumEvent as RawRumResourceEvent)._dd!
+      expect(traceInfo.rule_psr).toBeUndefined()
+    })
+
+    it('should define rule_psr to 0 if tracingSampleRate is set to 0', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.spec.ts:243]: **👏 praise:** ‏👍

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts [hunk 3]
```diff
@@ -167,5 +167,5 @@ function toPerformanceEntryRepresentation(entry: RumPerformanceEntry): Performan
  * @returns number between 0 and 1 which represents tracing sample rate
  */
 function getRulePsr(configuration: RumConfiguration) {
-  return configuration.tracingSampleRate / 100
+  return typeof configuration.tracingSampleRate === 'number' ? configuration.tracingSampleRate / 100 : undefined
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:170]: **🥜 nitpick:** ‏actually, we have a `isNumber` helper in `core/src/tools/utils`

**Review by bcaudan — APPROVED** [review]

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**